src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -655,20 +655,10 @@ private void getCanvasResponse(HttpServletRequest request,
 
 		String linkValue = null;
 
-		Header[] headers = canvasResponse.getAllHeaders();
-		for (Header header : headers) {
-			M_log.debug("Key : " + header.getName() 
-					+ " ,Value : " + header.getValue());
-			if(header.getName().equals("Link")){
-				linkValue = header.getValue();
-				break;
-			}
-		}
-
-		M_log.debug("LINK VALUE: " + linkValue);
+		linkValue = extractNextLink(canvasResponse, linkValue);
 
```

Comment:
What is the need to pass in the `linkValue` which is null to the `extractNextLink` when you can just create on in the `extractNextLink` method call.